release.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+REPOSITORY="origin"
+BRANCH_FROM="main"
+while getopts ":a:f:b:r:k:" opt; do
+  case $opt in
+    a) RELEASE_API_VERSION="${OPTARG}"
+    ;;
+    f) RELEASE_VERSION="${OPTARG}"
+    ;;
+    b) BRANCH_FROM="${OPTARG}"
+    ;;
+    r) REPOSITORY="${OPTARG}"
+    ;;
+    k) GPG_KEY="${OPTARG}"
+    ;;
+
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+
+  case ${OPTARG} in
+    -*) echo "Option $opt needs a valid argument"
+    exit 1
+    ;;
+  esac
+done
+
+if [[ -z "${GPG_KEY}" ]]; then
+    echo "GPG_KEY not set unable to sign the release. Please specify -k <YOUR_GPG_KEY>" 1>&2
+    exit 1
+else
+  export GPG_KEY
+fi
+
+git stash --all
+echo "Creating release branch from ${BRANCH_FROM}"
+git fetch -q "${REPOSITORY}"
+git branch prepare-release "${REPOSITORY}/${BRANCH_FROM}"
+
+if [[ -n ${RELEASE_API_VERSION} ]]; then
+  echo "Releasing Public APIs as ${RELEASE_API_VERSION}"
+  ./release-api.sh "${RELEASE_API_VERSION}"
```

Comment:
abort if this script fails? Or use `set -e`

Reply:
Double checking which shell option would cause it to abort was on my list, thanks for confirming.